src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -52,6 +52,21 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
       IntervalState{StateType::Released, empty_regiondata}
   };
 
+  bool is_reserve_operation = state == StateType::Reserved && !use_tag_inplace;
+  bool is_uncommit_operation = state == StateType::Reserved && use_tag_inplace;
+  bool is_commit_operation = state == StateType::Committed;
+  if (is_reserve_operation) {
+    stA.out.set_reserve_stack(metadata.stack_idx);
+    stB.in.set_reserve_stack(metadata.stack_idx);
+    stA.out.set_commit_stack(NativeCallStackStorage::invalid);
+    stA.in.set_commit_stack(NativeCallStackStorage::invalid);
+  }
+  if (is_commit_operation) {
+    stA.out.set_commit_stack(metadata.stack_idx);
+    stB.in.set_commit_stack(metadata.stack_idx);
+    stA.out.set_reserve_stack(NativeCallStackStorage::invalid);
+    stB.in.set_reserve_stack(NativeCallStackStorage::invalid);
+  }
```

Comment:
I'd prefer to see this as:

```
  // set default values, then depending on operation later, fill the required states
  stA.out.set_commit_stack(NativeCallStackStorage::invalid);
  stA.in.set_commit_stack(NativeCallStackStorage::invalid);
  stA.out.set_reserve_stack(NativeCallStackStorage::invalid);
  stB.in.set_reserve_stack(NativeCallStackStorage::invalid);

   if (is_reserve_operation) {
    stA.out.set_reserve_stack(metadata.stack_idx);
    stB.in.set_reserve_stack(metadata.stack_idx);
  }

  if (is_commit_operation) {
    stA.out.set_commit_stack(metadata.stack_idx);
    stB.in.set_commit_stack(metadata.stack_idx);
  }

```

Than later on we can skip setting these default values altogether and only touch the ones we need to.

Reply:
Done.